src/mvt_getter.py

```diff
@@ -0,0 +1,297 @@
+            groupby,
+            pixel_size_at_zoom(maxzoom, pxtolerance) ** 2,
+            minzoom,
+            x,
+            y,
+            ",".join([(("'coastline'" if name == 'natural' else 'null') + ' as "' + name + '"') for name in names]),
+            pixel_size_at_zoom(maxzoom, pxtolerance),
+            pixel_size_at_zoom(maxzoom, pxtolerance),
+            pixel_size_at_zoom(maxzoom, pxtolerance),
+            minzoom,
+            x,
+            y,
+            pixel_size_at_zoom(maxzoom, pxtolerance) ** 2
+        )
+    elif vec == "line":
+        query = """select ST_AsMVTGeom(way, ST_TileEnvelope(%s, %s, %s), 4096, 64, true) as %s, %s from
+                        (select (ST_Dump(ST_Multi(ST_Simplify(ST_LineMerge(way),%s)))).geom as %s, %s from
+                            (select ST_Union(way) as %s, %s from
+                                %s
+                                where (%s)
+                                and way && ST_TileEnvelope(%s, %s, %s)
+                            group by %s
+                            ) p
+                        ) p
+          """ % (
+            minzoom,
+            x,
+            y,
+            geomcolumn,
+            groupby,
+            pixel_size_at_zoom(maxzoom, pxtolerance),
+            geomcolumn,
+            groupby,
+            geomcolumn,
+            select,
+            table[vec],
+            adp,
+            minzoom,
+            x,
+            y,
+            ",".join(["tags->" + "'" + i + "'" for i in names]),
+        )
+    elif vec == "point":
+        query = """select ST_AsMVTGeom(way, ST_TileEnvelope(%s, %s, %s), 4096, 64, true) as %s, %s
+                        from %s where (%s)
+                        and way && ST_TileEnvelope(%s, %s, %s)
+                        limit 10000
+                 """ % (
+            minzoom,
+            x,
+            y,
+            geomcolumn,
+            select,
+            table[vec],
+            adp,
+            minzoom,
+            x,
+            y,
+        )
+
+    return query
+
+
+if __name__ == "__main__":
+    parser = OptionParser()
+    parser.add_option("-s", "--stylesheet", dest="filename", metavar="FILE")
+
+    (options, args) = parser.parse_args()
+
+    style = MapCSS(0, 30)
+    style.parse(filename=options.filename)
+
+    zooms = [
+        (1, 2),
+        (2, 3),
+        (3, 4),
+        (4, 5),
+        (5, 6),
+        (6, 7),
+        (7, 8),
+        (8, 9),
+        (9, 10),
+        (10, 11),
+        (11, 12),
+        (12, 13),
+        (13, 14),
+        (14, 30)
+    ]
+
+    for (minzoom, maxzoom) in zooms:
+        print(
+            """create or replace function public.basemap_z%s(x integer, y integer)
+        returns bytea
+        as $$
+        declare
+            area_mvt bytea;
+            line_mvt bytea;
+            node_mvt bytea;
+            result bytea;
+        begin
+            select into area_mvt
```

Comment:
this can be language sql function and that will save some cycles

select (select ST_... ) || (select ST_ ...) || (select ST_ ...)